### ISSUE TITLE
Updated version

### ISSUE THREAD

**Issue by Chreece** [TheDigital1/ESP32_BTHome#5]
Thank you very much for your time spending on this project. It really helped me to understand the BTHome protocol.
In tried to use the code as an example and unfortunately it didn't work so I tried with fully respect to update it.
Here you can find an updated version:
https://github.com/Chreece/BTHomeV2-ESP32-example

I tried to make it more user friendly, using a header file with variables for the Object IDs and boolean values.
Also added an auto length calculation.

The only thing that needs to be changed for an implemantation is the device_name and the payload for the entities.

Unfortunately I haven't also understand the encryption documentation and couldn't implement it but I will keep up on it...

I hope you like the new version of your code :)

Keep up!

**Comment by TheDigital1** [comment]
Thanks Chreece!
My goal of uploading my example code was just to give people a quick starting point for BTHome on ESP32's.
I had hopes of evolving it and adding more examples and functionality, however I have not had time. So I greatly appreciate you picking it up and continuing the work on it.

I have put a link to your project on my main page.

I guess I will keep my project up for old BTHome V1, and yours can focus on V2

Thanks again!

**Comment by Chreece** [comment]
Great attidute man!

I'm not a programmer, just a hobbist. I had in mind to try build a DIY Bluetooth people counter and because of that I stubled upon  your project. It is a start for everyone who wants to build a DIY BT sensor.

In that in mind i try to made the code simplier and just now I uploaded the V2 with minimal user input.

It just needs the device name, the number of the sensors (entities) that will be send and the states of them. The rest is done under the hood and not in the .ino file that was edited by the user.

All of that because I'm lazy and need a simply interface to build my feature sensors.

The V3 of the example should also have encryption if I will understand it or if chatgtp helps me :)

Keep Up!
